src/python/pants/core/goals/generate_lockfiles.py

```diff
@@ -362,6 +368,16 @@ def activated(cls, union_membership: UnionMembership) -> bool:
             """
         ),
     )
+    diff_generated = BoolOption(
```

Comment:
This should probably be called `generate_diff`?

Reply:
fine by me :)

Reply:
Ansible uses `--diff` - could we use that here?

"Generated" and "generate" do not add any additional info on the CLI, because the goal is already "generate-lockfiles". So, I would prefer short and sweet: `./pants generate-lockfiles --diff --resolve=foobar`.

Reply:
perhaps worth considering also what this would look like if the goal name was just `lock`..?
`./pants lock --diff --resolve=foobar` , does that still make sense?
Context: #15568

Reply:
Yes. A shorter arg name aligns well with the purpose of changing the goal name.

In `lock --diff`, diff what? Diff the lock.

I think that works very well.